flink-cdc-runtime/src/main/java/com/ververica/cdc/runtime/operators/transform/RowFilter.java

```diff
@@ -0,0 +1,111 @@
+import com.ververica.cdc.runtime.parser.JaninoParser;
+import com.ververica.cdc.runtime.parser.TransformParser;
+import com.ververica.cdc.runtime.typeutils.DataTypeConverter;
+import org.codehaus.janino.ExpressionEvaluator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+/** The RowFilter applies to describe the row filter of filtering tables. */
+public class RowFilter {
+    private static final Logger LOG = LoggerFactory.getLogger(RowFilter.class);
+    private final String expression;
+    private final String scriptExpression;
+    private final List<String> columnNames;
+    private ExpressionEvaluator expressionEvaluator;
+
+    public RowFilter(String expression, String scriptExpression, List<String> columnNames) {
+        this.expression = expression;
+        this.scriptExpression = scriptExpression;
+        this.columnNames = columnNames;
+    }
+
+    public static RowFilter of(
+            String expression, String scriptExpression, List<String> columnNames) {
+        return new RowFilter(expression, scriptExpression, columnNames);
+    }
+
+    public static Optional<RowFilter> generateRowFilter(String filterExpression) {
+        if (StringUtils.isNullOrWhitespaceOnly(filterExpression)) {
+            return Optional.empty();
+        }
+        List<String> columnNames = TransformParser.parseFilterColumnNameList(filterExpression);
+        String scriptExpression =
+                TransformParser.translateFilterExpressionToJaninoExpression(filterExpression);
+        return Optional.of(of(filterExpression, scriptExpression, columnNames));
+    }
+
+    public boolean run(BinaryRecordData after, TableInfo tableInfo) {
+        if (expressionEvaluator == null) {
+            cacheExpressionEvaluator(tableInfo);
+        }
+        try {
+            return (Boolean) expressionEvaluator.evaluate(generateParams(after, tableInfo));
+        } catch (InvocationTargetException e) {
+            LOG.error("Table:{} filter:{} execute failed. {}", tableInfo.getName(), expression, e);
+            throw new RuntimeException(e);
+        }
+    }
+
+    private Object[] generateParams(BinaryRecordData after, TableInfo tableInfo) {
+        List<Column> columns = tableInfo.getSchema().getColumns();
+        List<Object> params = new ArrayList<>();
+        RecordData.FieldGetter[] fieldGetters = tableInfo.getFieldGetters();
+        for (int i = 0; i < columns.size(); i++) {
+            if (columnNames.contains(columns.get(i).getName())) {
+                params.add(
+                        DataTypeConverter.convertToOriginal(
+                                fieldGetters[i].getFieldOrNull(after), columns.get(i).getType()));
+            }
+        }
+        return params.toArray();
+    }
+
+    private void cacheExpressionEvaluator(TableInfo tableInfo) {
```

Comment:
Could we cache the generated ExpressionEvaluator in transform operator where we can build cache or clear the cache in open() and close() methods?